pulp_rpm/app/tasks/publishing.py

```diff
@@ -113,6 +114,7 @@ def publish_artifacts(self, content):
             relative_path = content_artifact.relative_path
             if content_artifact.content.pulp_type == Package.get_pulp_type():
                 relative_path = os.path.join(
+                    prefix,
```

Comment:
sometimes the repo could contain: `../`
example:
```
[variant-AppStream]
id = AppStream
name = AppStream
packages = ../../../AppStream/x86_64/kickstart/Packages
repository = ../../../AppStream/x86_64/kickstart/
type = variant
uid = AppStream
```
from: http://mirror.linux.duke.edu/pub/centos/8/BaseOS/x86_64/kickstart/.treerepo
I'm wondering what would happen with this prefix for this case

Reply:
The publication that contains a sub-repo always has the subrepo published inside the root of the publication. That means the paths in the original treeinfo file are no longer valid. A new treeinfo file is generated here[0]. It is based on the changes to the TreeInfo object made here[1].

[0] https://github.com/pulp/pulp_rpm/pull/1755/files#diff-7a569a289e63f956d6451ebe97f33a96R417
[1] https://github.com/pulp/pulp_rpm/pull/1755/files#diff-ca4739239d72f5108b090db97078ce70R401

Reply:
Oh! I didn't notice [0] before, this is pretty cool!